File: 59.R
# XOR decryption

# Constants
CIPHER_URL <- 'https://projecteuler.net/project/resources/p059_cipher.txt'

decrypt <- function(cipher) {
    for ( key1 in seq.int(97, 122) ) {
        for ( key2 in seq.int(97, 122) ) {
            for ( key3 in seq.int(97, 122) ) {
                key <- c(key1, key2, key3)
                plain_ascii <- bitwXor(cipher, key)
                plain <- rawToChar(as.raw(plain_ascii))
                if ( grepl(' the ', plain) ) {
                    return(sum(plain_ascii))
                }
            }
        }    
    }
}

cipher <- suppressWarnings(read.csv(sub('https', 'http', CIPHER_URL),
                                    header=FALSE, colClasses='character',
                                    na.strings=""))
cipher <- as.integer(cipher[1,])

cat(decrypt(cipher), fill=TRUE)

File: 59.js
#!/usr/bin/env node

// XOR decryption

var CIPHER_URL = 'https://projecteuler.net/project/resources/p059_cipher.txt';

var request = require('request');
request(CIPHER_URL, function(error, response, body) {
    if (!error && response.statusCode == 200) {
        decrypt(body.split(","));
    }
});

function decrypt(cipher) {
    for (var key1 = 97; key1 < 123; key1++) {
        for (var key2 = 97; key2 < 123; key2++) {
            for (var key3 = 97; key3 < 123; key3++) {
                var key = [key1, key2, key3];
                var plain = '';
                var sum = 0;
                for (var i = 0; i < cipher.length; i++) {
                    var xor = parseInt(cipher[i]) ^ key[i % 3];
                    plain += String.fromCharCode(xor);
                    sum += xor;
                }
                if (plain.indexOf(' the ') >= 0) {
                    console.log(sum);
                    return;
                }
            }
        }
    }
}
